store/mockstore/unistore/tikv/mvcc.go

```diff
@@ -1386,14 +1408,32 @@ func (store *MVCCStore) DeleteFileInRange(start, end []byte) {
 	store.db.DeleteFilesInRange(start, end)
 }
 
+// Get implements the MVCCStore interface.
+func (store *MVCCStore) Get(reqCtx *requestCtx, key []byte, version uint64) ([]byte, error) {
+	lockPairs, err := store.CheckKeysLock(version, reqCtx.rpcCtx.ResolvedLocks, reqCtx.rpcCtx.CommittedLocks, key)
+	if err != nil {
+		return nil, err
+	}
+	if len(lockPairs) != 0 {
+		return getValueFromLock(lockPairs[0].lock), nil
+	}
+	val, err := reqCtx.getDBReader().Get(key, version)
+	return safeCopy(val), err
+}
+
 // BatchGet implements the MVCCStore interface.
 func (store *MVCCStore) BatchGet(reqCtx *requestCtx, keys [][]byte, version uint64) []*kvrpcpb.KvPair {
 	pairs := make([]*kvrpcpb.KvPair, 0, len(keys))
 	remain := make([][]byte, 0, len(keys))
 	for _, key := range keys {
-		err := store.CheckKeysLock(version, reqCtx.rpcCtx.ResolvedLocks, key)
+		lockPairs, err := store.CheckKeysLock(version, reqCtx.rpcCtx.ResolvedLocks, reqCtx.rpcCtx.CommittedLocks, key)
 		if err != nil {
 			pairs = append(pairs, &kvrpcpb.KvPair{Key: key, Error: convertToKeyError(err)})
+		} else if len(lockPairs) != 0 {
+			value := getValueFromLock(lockPairs[0].lock)
+			if value != nil {
```

Comment:
If the lock type is not `Put`, value will be nil, so we skip this key here.

Reply:
The lock type is either `Put` or `Del` because only locks that can block read are in `LockPair`.

Reply:
I mistakenly thought the `Del` record needs a `nil` return, LGTM.